uniffi_bindgen/src/lib.rs

```diff
@@ -311,6 +457,34 @@ impl From<&ComponentInterface> for Config {
     }
 }
 
+impl <'a, C: From<&'a ComponentInterface>> From<&'a ComponentInterface> for BackendConfig<C> {
+    fn from(ci: &'a ComponentInterface) -> Self {
+        BackendConfig {
+            bindings: ci.into(),
+        }
+    }
+}
+
+impl <C: BindingGeneratorConfig> BackendConfig<C> {
+    fn load_config<P: AsRef<Path>>(component: &ComponentInterface, path: P) -> Result<C> {
+        let default_config: Self = component.into();
+        let path = path.as_ref();
+        let contents = slurp_file(&path)
+            .with_context(|| format!("Failed to read config file from {:?}", &path))?;
+        let loaded_config: Self = toml::de::from_str(&contents)
+            .with_context(|| format!("Failed to generate config from file {:?}", &path))?;
+        Ok(loaded_config.merge_with(&default_config).bindings)
+    }
+}
```

Comment:
I'm not understanding the relationship between `BindingGeneratorConfig` and `BackendConfig`.  Is there a way to simplify this?  If not, I think some docstrings would definitely be useful.

Reply:
I dramatically simplified this in #1171, the `BackendConfig` existed only so the configuration can be under the `bindings` key.. but that's unnecessary complexity 😄 I got rid of that type there so now it's only the trait